api/user.go

```diff
@@ -2538,3 +2540,27 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, GetProtocol(r)+"://"+r.Host, http.StatusFound)
 	}
 }
+
+func userTyping(req *model.WebSocketRequest, responseData map[string]interface{}) *model.AppError {
+	var ok bool
+	var teamId string
+	if teamId, ok = req.Data["team_id"].(string); !ok || len(teamId) != 26 {
```

Comment:
This won't work for cross-team DMs, but I just tested it and that happens with the existing code anyway.